File: app/src/main/java/net/radityalabs/contactapp/presentation/presenter/BasePresenter.java
package net.radityalabs.contactapp.presentation.presenter;

import net.radityalabs.contactapp.presentation.view.BaseView;

/**
 * Created by radityagumay on 4/11/17.
 */

public interface BasePresenter<T extends BaseView>{

    void attachView(T view);

    void detachView();
}

File: app/src/main/java/net/radityalabs/contactapp/presentation/presenter/BasePresenter.kt
package net.radityalabs.contactapp.presentation.presenter

import net.radityalabs.contactapp.presentation.view.BaseView

/**
 * Created by radityagumay on 4/11/17.
 */

interface BasePresenter<T : BaseView> {

    fun attachView(view: T)

    fun detachView()
}
